packages/wonder-blocks-data/src/hooks/use-shared-cache.js

```diff
@@ -0,0 +1,106 @@
+// @flow
+import * as React from "react";
+import {KindError, Errors} from "@khanacademy/wonder-stuff-core";
+import {InMemoryCache} from "../util/in-memory-cache.js";
+import type {ValidData} from "../util/types.js";
+
+/**
+ * A function for inserting a value into the cache or clearing it.
+ */
+type CacheValueFn<TValue: ValidData> = (value: ?TValue) => void;
+
+/**
+ * This is the cache.
+ * It's incredibly complex.
+ * Very in-memory. So cache. Such complex. Wow.
+ */
+const cache = new InMemoryCache();
```

Comment:
Something we may want to explore in the future is providing a way to swap out cache implementations.  During development it may be useful to debug/test with a cache that supports time-travel debugging by storing values using immutable-js (or something like that) so that devs could easily move between different cache states.  No action req'd.  Just an idle thought.

Reply:
Yes. WB Data used to have a much more complex, defined caching API. I removed it because ultimately, we didn't in fact use it like we thought we would.

I was trying to envisage a scenario where we might want the cache to be in a specific state before starting a test and I wasn't able to convince myself that folks shouldn't do the right behavior to get it to that state rather than just jump there.

However, I think real world pain will illustrate it clearly and we can then act accordingly.